### DISCUSSION TITLE
Is it possible to create expenses and working time via script?

### DISCUSSION THREAD

**Discussion by techsupportkoeln** [kimai/kimai#3653] [General]
I have every month around 30 expenses and working times to create manually at the end of the month.
Example:

Customer 1, Expenses, Cloud-Server, 01.11.-30.11.2022, 999 €

Is it possible to create this via batch file or something else over the API? The only thing which change from moth to month is the date in the description.

**Comment by kevinpapst** [comment]
Yes, there are APIs for Timesheet and Expense

**Comment by techsupportkoeln** [comment]
not that easy to implement this for me. But i will take a look on the API at the end of this year.

if i open "https://my-kimai-url/en/timesheet/create" then i have an empty create mask for workingtime. Is it possible to open this mask with values over this link, as example: https://my-kimai-url/en/timesheet/create#Customer1#Project1#Activity1#Description1#Tag1 or is the only way over the API?

I use Zammad as Ticketsystem and i look for short ways to create working times without stupid copy & paste actions.

**Reply by kevinpapst** [reply]
Install Postman, import Kimai API, fill in fields => done.
Not much knowledge required. Copy & paste the Curl call. Adjust it for every customer. Put it in a batch file. Run it every month.
Not sure about the Windows world though.

Or you can simply use ?project=ID&activity=ID... in the create URL.

**Reply by techsupportkoeln** [reply]
thx for this help. Is it correct that the URL only works for the timesheet?

With https://<kimai-url>/en/expenses/create?description=test i got only an empty expenses mask.

**Reply by kevinpapst** [reply]
Yes, likely correct if it doesn't work with expenses.
